app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java

```diff
@@ -89,55 +92,31 @@ private static void addEvents(Context c, Uri uri) {
-
-                // Get start and end time
-                long startMillis = Utils.getDateTime(strStart).getTime();
-                long endMillis = Utils.getDateTime(strEnd).getTime();
-
-                ContentValues values = new ContentValues();
-
-                // Put the received values into a contentResolver to
-                // transmit the to Google Calendar
-                values.put(CalendarContract.Events.DTSTART, startMillis);
-                values.put(CalendarContract.Events.DTEND, endMillis);
-                values.put(CalendarContract.Events.TITLE, title);
-                values.put(CalendarContract.Events.DESCRIPTION, description);
-                values.put(CalendarContract.Events.CALENDAR_ID, id);
-                values.put(CalendarContract.Events.EVENT_LOCATION, location);
-                values.put(CalendarContract.Events.EVENT_TIMEZONE, R.string.calendarTimeZone);
-                contentResolver.insert(CalendarContract.Events.CONTENT_URI, values);
-            }
+        CalendarDao calendarDao = TcaDb.getInstance(c).calendarDao();
+
+        List<CalendarItem> calendarItems = calendarDao.getAllNotCancelled();
+        for (CalendarItem calendarItem: calendarItems) {
+            ContentValues values = new ContentValues();
+
+            // Put the received values into a contentResolver to
+            // transmit the to Google Calendar
+            values.put(CalendarContract.Events.DTSTART, Utils.getDateTime(calendarItem.getDtstart()).getTime());
+            values.put(CalendarContract.Events.DTEND, Utils.getDateTime(calendarItem.getDtend()).getTime());
+            values.put(CalendarContract.Events.TITLE, calendarItem.getTitle());
+            values.put(CalendarContract.Events.DESCRIPTION, calendarItem.getDescription());
+            values.put(CalendarContract.Events.CALENDAR_ID, id);
+            values.put(CalendarContract.Events.EVENT_LOCATION, calendarItem.getLocation());
+            values.put(CalendarContract.Events.EVENT_TIMEZONE, R.string.calendarTimeZone);
+            contentResolver.insert(CalendarContract.Events.CONTENT_URI, values);
         }
     }
 
-    /**
-     * Returns all stored events from db
-     *
-     * @return Cursor with all calendar events. Columns are
-     * (nr, status, url, title, description, dtstart, dtend, location)
-     */
-    private Cursor getAllFromDb() {
-        return db.rawQuery("SELECT * FROM calendar WHERE status!='CANCEL'", null);
-    }
-
-    public Cursor getFromDbForDate(Date date) {
+    public List<CalendarItem> getFromDbForDate(Date date) {
         // Format the requested date
         String requestedDateString = Utils.getDateString(date);
 
         // Fetch the data
-        return db.rawQuery("SELECT * FROM calendar WHERE dtstart LIKE ? AND status!='CANCEL' ORDER BY dtstart ASC", new String[]{"%" + requestedDateString + "%"});
+        return calendarDao.getAllByDateNotCancelled("%" + requestedDateString + "%");
```

Comment:
The SQL `%` wildcards should move to the DAO.

Also: I think we have a Date to String mapper for the database, so it should be possible to call `getAllByDateNotCancelled()` directly with a Date?

Reply:
Actuallly, I remembered that converter converts to minute accuracy, whereas it is not the case for this query, which typically is used to retrieve day's events. That is: converter would make a string "2018-01-18 10:11", whereas previously it was expected that the string would be "2018-01-18" which would return all the calendar items for the day.

I've restored the version where a string is passed, but moved the `%` into the query

Reply:
hmm, I think we'd should fix the query then to work with full-accuracy datetimes, e.g. by surrounding the parameter with [`date()`](https://www.sqlite.org/lang_datefunc.html), so we don't need to do the string conversion